votor/src/consensus_pool/vote_pool.rs

```diff
@@ -2,239 +2,375 @@ use {
     crate::common::Stake,
     solana_hash::Hash,
     solana_pubkey::Pubkey,
-    solana_votor_messages::consensus_message::VoteMessage,
-    std::collections::{BTreeMap, BTreeSet},
+    solana_votor_messages::{
+        consensus_message::VoteMessage,
+        vote::{Vote, VoteType},
+    },
+    std::collections::{btree_map::Entry, BTreeMap},
+    thiserror::Error,
 };
 
-/// There are two types of vote pools:
-/// - SimpleVotePool: Tracks all votes of a specfic vote type made by validators for some slot N, but only one vote per block.
-/// - DuplicateBlockVotePool: Tracks all votes of a specfic vote type made by validators for some slot N,
-///   but allows votes for different blocks by the same validator. Only relevant for VotePool's that are of type
-///   Notarization or NotarizationFallback
-pub(super) enum VotePool {
-    SimpleVotePool(SimpleVotePool),
-    DuplicateBlockVotePool(DuplicateBlockVotePool),
+#[derive(Debug, Error, PartialEq)]
```

Comment:
```suggestion
#[derive(Debug, Error, PartialEq, Eq)]
```

Reply:
nit: also, I would put `Error` at the end b/c it's not an `std` derive in this case, but not sure how this is usually done in agave